sympy/matrices/matrices.py

```diff
@@ -1547,6 +1547,10 @@ def _matrix_pow_by_jordan_blocks(self, num):
         def jordan_cell_power(jc, n):
             N = jc.shape[0]
             l = jc[0, 0]
+            if l == 0 and (num < 0) == True:
+                raise ValueError("Matrix det == 0; not invertible")
+            elif l == 0 and jc.rows > 1 and (n >= 0) == True and not n % 1 == 0:
```

Comment:
Is `(n >= 0) == True` necessary when `n < 0` is checked above?
`not n % 1 == 0` can be written `n % 1 != 0`.